src/util/ajax.js

```diff
@@ -151,7 +151,12 @@ function makeFetchRequest(requestParameters: RequestParameters, callback: Respon
                 return finishRequest(response, cacheableResponse, requestTime);
 
             } else {
-                return callback(new AJAXError(response.statusText, response.status, requestParameters.url));
+                response.text().then(text => {
```

Comment:
Would you consider attempting to read `response.text()` only when `statusText` is empty?